keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/DebianArtifactSupplier.kt

```diff
@@ -62,32 +63,47 @@ class DebianArtifactSupplier(
 
   override fun getVersionDisplayName(artifact: PublishedArtifact): String {
     // TODO: Frigga and Rocket version parsing are not aligned. We should consolidate.
-    val appversion = AppVersion.parseName(artifact.version)
-    return if (appversion?.version != null) {
-      appversion.version
-    } else {
-      artifact.version.removePrefix("${artifact.name}-")
+    return try {
+      val appversion = AppVersion.parseName(artifact.version)
```

Comment:
Since the same try/catch/log.warn logic appears in three separate places in this class, I'd create a helper function here to call in place of `AppVersion.parseName`, something like:

```kotlin
fun parseName(artifact: PublishedArtifact): AppVersion? =
  try {
    appVersion.parseName(artifact.version)
 } catch (ex: Exception) {
    log.warn("trying to parse artifact ${artifact.name} with version ${artifact.version} but got an exception", ex)
    null
}
```

That way, it's easier to change the logging statement in the future since it's only in one place.

Reply:
done